lib/core/proxy.c

```diff
@@ -467,6 +467,7 @@ static h2o_http1client_body_cb on_head(h2o_http1client_t *client, const char *er
     /* copy the response (note: all the headers must be copied; http1client discards the input once we return from this callback) */
     req->res.status = status;
     req->res.reason = h2o_strdup(&req->pool, msg.base, msg.len).base;
+    req->res.headers = (h2o_headers_t){NULL, 0, 0};
```

Comment:
Do we need to reset fields other than `size`?

Consider the case where the server sends an informational response containing 50 header fields, and then sending a final response containing the same number of header fields. I would prefer not to expand the buffer twice to store the headers. This is a weak preference, but I do not see why it can simply be done by not resetting the other fields.

Reply:
> I would prefer not to expand the buffer twice to store the headers.

No, when sending 1xx we're not allocating or expanding any vectors or buffers

https://github.com/h2o/h2o/blob/b43efc484282529f567045543898b203b2a089f9/lib/core/proxy.c#L578

If there're no performance reasons, I'd prefer to clear whole struct explicitly rather than leave broken pointer (req->res.headers.entries).

Reply:
Or, we might should clear req->res.headers at the end of http1's finalostream_send_informational..

Reply:
IMO it would depend on how an output filter (e.g. the headers handler) initiates Early Hints, when a generator that does not immediately return a final response is used.

Clearing `req->res.headers` would be fine if `h2o_send_informational` is going to be used for such purposes as well. Otherwise, we should clear the field in each protocol implementation, at the moment they serialize the headers to send them.

Reply:
> Clearing req->res.headers would be fine if h2o_send_informational is going to be used for such purposes as well.
Yes, I think h2o_send_informational will be used for such use cases. https://github.com/h2o/h2o/pull/1727/commits/d6e1e3b15503930f581f2d36eeb645695debc5c7 made it clear req->res